src/fshtypes/rules/Rule.ts

```diff
@@ -1,5 +1,5 @@
 // TODO: Is this really an Interface
```

Comment:
In looking at this more, I think I'm solidifying my opinion on this unless you have an opinion to the contrary.
Interfaces seem like they're more commonly used as a "contract" to define rules that other classes must follow. They are beneficial in the sense that you can have a class implement multiple interfaces to have multiple contracts to follow.
In our specific case, it seems like it would be more beneficial to have Rule be an abstract class from which the other rule types can inherit. That way we are binding the rule class to a specific set of requirements, but don't have to explicitly define them within each rule class.

Reply:
I think I'm inclined to agree at this point.  I think I initially did it this way because there's not really much going on there (no actual _logic_ that needs to be inherited -- just properties).  But... I foresee a future in which rules may have some additional features added -- and it would be a pain to have to add them in every separate rule class!  So... good call.  I'll make the change.

Reply:
I'm also going to evaluate if we really need the `ruleType` property.  I'm sure I had a reason for adding it before, but I'm not so sure what that reason is anymore.  If I confirm we don't need it, then I'm going to remove it.

Reply:
OK -- it looks like rule types are not needed, so I have removed them (will be in next PR).